### PR TITLE
Fix coverage data file path when build is done on a different machine.

### PR DESCRIPTION
**Pull request by mazimkhan**

### CODE REVIEW

File: mbed_greentea/mbed_test_api.py [hunk 1]
```diff
@@ -334,12 +373,15 @@ def get_auto_property_value(property_name, line):
...
                     coverage_end_line = line[:idx]  # Coverage payload
-                    cov_data = {
-                        "path" : coverage_start_data[1],
-                        "payload" : coverage_end_line,
-                        "encoding" : 'hex'  # hex, base64
-                    }
-                    coverage_start_data_list.append(cov_data)
+                    cov_data_file_path = fix_coverage_report_path(coverage_start_data[1])
+
+                    if cov_data_file_path is not None:
```
Comment (PrzemekWirkus) [inline mbed_greentea/mbed_test_api.py]:
```python
if cov_data_file_path is not None:
```
is the same as:
```python
if cov_data_file_path:
```

File: mbed_greentea/mbed_test_api.py [hunk 2]
```diff
@@ -230,6 +230,45 @@ def get_auto_property_value(property_name, line):
...
+        """
+        fixed_path = reported_path
+        cov_data_file_name = os.path.basename(reported_path)
+        cov_data_file_dir = os.path.dirname(reported_path)
+        if not os.path.exists(cov_data_file_dir):
+            fixed_path = None
+            gt_logger.gt_log_warn("gcov data file path reported by target does not exist!")
+            # Try fixing path by replacing reported path upto current working dir name with current working dir path
+            while True:
+                pos = cov_data_file_dir.find('build')
```
Comment (PrzemekWirkus) [inline mbed_greentea/mbed_test_api.py]: How about my working directory is in:
```
c:\build\build\build\mymodule
```
?

I think you get my point :D

Comment (mazimkhan) [inline mbed_greentea/mbed_test_api.py]: You are overlooking the while loop and check if the constructed path exists

**Comment by mazimkhan** [comment]
Closing as this is now done by compiling with relative path flag. see PR https://github.com/ARMmbed/target-mbed-gcc/pull/33

**Pull request closed without merging**